packages/mobile/src/services/offline-downloader/offline-sync.ts

```diff
@@ -156,32 +166,20 @@ export const syncCollectionTracks = async (
   // TODO: will discovery serve a removed playlist?
   if (!updatedCollection) return
 
-  if (
-    moment(updatedCollection.updated_at).isSameOrBefore(
-      offlineCollection.updated_at
-    )
-  ) {
-    return
-  }
-
-  downloadCollection(
-    updatedCollection,
-    isFavoritesDownload,
-    /* skipTracks */ true
-  )
-  if (updatedCollection.cover_art_sizes !== offlineCollection.cover_art_sizes) {
-    downloadCollectionCoverArt(updatedCollection)
-  }
-
-  const oldTrackIds = new Set(
-    offlineCollection.tracks?.map((track) => track.track_id)
-  )
+  const oldTrackIds = new Set([
+    ...(offlineCollection.tracks
+      ?.map((track) => track.track_id)
+      .filter((trackId) => downloadedTracks[trackId] != null) || []),
+    ...queuedTracks
+  ])
   const newTrackIds = new Set(
     updatedCollection.tracks?.map((track) => track.track_id)
   )
+
   const addedTrackIds = [...newTrackIds].filter(
```

Comment:
dont need a spread here cause filter creates new array!

Reply:
need it because its a set :P

Reply:
Oh yeah!